tests/test_ctx_options.cpp

```diff
@@ -30,15 +30,108 @@
+
+bool is_allowed_to_raise_priority()
+{
+    // NOTE1: if setrlimit() fails with EPERM, this means that current user has not enough permissions.
+    // NOTE2: even for privileged users (e.g., root) getrlimit() would usually return 0 as nice limit; the only way to
+    //        discover if the user is able to increase the nice value is to actually try to change the rlimit:
+    struct rlimit rlim;
+    rlim.rlim_cur = 40;
+    rlim.rlim_max = 40;
+    if (setrlimit(RLIMIT_NICE, &rlim) == 0)
+    {
+        // rlim_cur == 40 means that this process is allowed to set a nice value of -20
+        if (WAIT_FOR_BACKGROUND_THREAD_INSPECTION)
+            printf ("This process has enough permissions to raise ZMQ background thread priority!\n");
+        return true;
+    }
+
+    if (WAIT_FOR_BACKGROUND_THREAD_INSPECTION)
+        printf ("This process has NOT enough permissions to raise ZMQ background thread priority.\n");
+    return false;
+}
+
+#else
+
+#define TEST_POLICY                                       (0)
+
+bool is_allowed_to_raise_priority()
+{
+    return false;
+}
+
+#endif
+
+
+void test_ctx_thread_opts(void* ctx)
+{
+    int rc;
+
+    // verify that setting negative values (e.g., default values) fail:
+    rc = zmq_ctx_set(ctx, ZMQ_THREAD_SCHED_POLICY, ZMQ_THREAD_SCHED_POLICY_DFLT);
+    assert (rc == -1 && errno == EINVAL);
+    rc = zmq_ctx_set(ctx, ZMQ_THREAD_PRIORITY, ZMQ_THREAD_PRIORITY_DFLT);
+    assert (rc == -1 && errno == EINVAL);
+    rc = zmq_ctx_set(ctx, ZMQ_THREAD_AFFINITY, ZMQ_THREAD_AFFINITY_DFLT);
+    assert (rc == -1 && errno == EINVAL);
+
+
+    // test scheduling policy:
+
+    // set context options that alter the background thread CPU scheduling/affinity settings;
+    // as of ZMQ 4.2.3 this has an effect only on POSIX systems (nothing happens on Windows, but still it should return success):
+    rc = zmq_ctx_set(ctx, ZMQ_THREAD_SCHED_POLICY, TEST_POLICY);
+    assert (rc == 0);
+
+
+    // test priority:
+
+    // in theory SCHED_OTHER supports only the static priority 0 but quoting the docs
+    //     http://man7.org/linux/man-pages/man7/sched.7.html
+    // "The thread to run is chosen from the static priority 0 list based on
+    // a dynamic priority that is determined only inside this list.  The
+    // dynamic priority is based on the nice value [...]
+    // The nice value can be modified using nice(2), setpriority(2), or sched_setattr(2)."
+    // ZMQ will internally use nice(2) to set the nice value when using SCHED_OTHER.
+    // However changing the nice value of a process requires appropriate permissions...
+    // check that the current effective user is able to do that:
+    if (is_allowed_to_raise_priority())
+    {
+        rc = zmq_ctx_set(ctx, ZMQ_THREAD_PRIORITY, 1 /* any positive value different than the default will be ok */);
+        assert (rc == 0);
+    }
+
+
+    // test affinity:
```

Comment:
Given this option is new and in DRAFT state, you should ifdef it:

```
../tests/test_ctx_options.cpp: In function ‘void test_ctx_thread_opts(void*)’:
../tests/test_ctx_options.cpp:84:27: error: ‘ZMQ_THREAD_AFFINITY’ was not declared in this scope
     rc = zmq_ctx_set(ctx, ZMQ_THREAD_AFFINITY, ZMQ_THREAD_AFFINITY_DFLT);
                           ^
../tests/test_ctx_options.cpp:84:48: error: ‘ZMQ_THREAD_AFFINITY_DFLT’ was not declared in this scope
     rc = zmq_ctx_set(ctx, ZMQ_THREAD_AFFINITY, ZMQ_THREAD_AFFINITY_DFLT);
```

\#ifdef ZMQ_THREAD_AFFINITY
...
\#endif